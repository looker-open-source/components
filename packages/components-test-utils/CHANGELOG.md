# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
