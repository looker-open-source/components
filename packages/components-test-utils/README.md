# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
