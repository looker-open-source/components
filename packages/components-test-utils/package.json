{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
