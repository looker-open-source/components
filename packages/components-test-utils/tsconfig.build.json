{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
