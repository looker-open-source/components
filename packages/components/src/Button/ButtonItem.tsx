/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { forwardRef, MouseEvent, Ref, useContext, useState } from 'react'
import styled from 'styled-components'
import {
  CompatibleHTMLProps,
  space,
  SpaceProps,
  omitStyledProps,
} from '@looker/design-tokens'
import { inputHeight } from '../Form/Inputs/height'
import { ButtonSetContext } from './ButtonSetContext'

export interface ButtonItemProps
  extends SpaceProps,
    Omit<CompatibleHTMLProps<HTMLButtonElement>, 'type' | 'aria-pressed'> {
  value?: string
  focusVisible?: boolean
}

const ButtonLayout = forwardRef(
  (
    { children, onClick, value, onBlur, onKeyUp, ...props }: ButtonItemProps,
    ref: Ref<HTMLButtonElement>
  ) => {
    const { disabled, value: contextValue, onItemClick } = useContext(
      ButtonSetContext
    )

    const [isFocusVisible, setFocusVisible] = useState(false)

    const handleOnKeyUp = (event: React.KeyboardEvent<HTMLButtonElement>) => {
      setFocusVisible(true)
      onKeyUp && onKeyUp(event)
    }

    const handleOnBlur = (event: React.FocusEvent<HTMLButtonElement>) => {
      setFocusVisible(false)
      onBlur && onBlur(event)
    }

    function handleClick(e: MouseEvent<HTMLButtonElement>) {
      onClick && onClick(e)
      if (!e.defaultPrevented) {
        onItemClick && onItemClick(e)
      }
    }

    const itemValue =
      value !== undefined ? value : typeof children === 'string' ? children : ''

    const selected = contextValue
      ? typeof contextValue === 'string'
        ? contextValue === itemValue
        : contextValue.includes(itemValue)
      : false

    return (
      <ButtonOuter
        aria-pressed={selected}
        ref={ref}
        onClick={handleClick}
        value={itemValue}
        disabled={disabled}
        focusVisible={isFocusVisible}
        onKeyUp={handleOnKeyUp}
        onBlur={handleOnBlur}
        {...omitStyledProps(props)}
      >
        {children}
      </ButtonOuter>
    )
  }
)

ButtonLayout.displayName = 'ButtonLayout'

const ButtonOuter = styled.button<ButtonItemProps>`
  ${(props) =>
    props.focusVisible &&
    `box-shadow: 0 0 0.5px 1px ${props.theme.colors.keyFocus}`}
`

export const ButtonItem = styled(ButtonLayout)`
  align-items: center;
  background: transparent;
  border: none;
  color: ${({ theme }) => theme.colors.text3};
  cursor: pointer;
  display: inline-flex;
  font-family: ${({ theme }) => theme.fonts.brand};
  font-size: ${({ theme }) => theme.fontSizes.small};
  height: ${inputHeight};
  justify-content: center;
  margin: 0;
  padding: 0 ${({ theme }) => theme.space.small};
  transition: background ${({ theme }) => theme.transitions.quick}ms ease;
  user-select: none;

  ${space}

  &:active,
  &:focus,
  &:hover {
    background: ${({ theme }) => theme.colors.neutralSubtle};
    color: ${({ theme }) => theme.colors.text5};
  }

  &:focus {
    outline: none;
  }

  &[disabled] {
    color: ${(props) => props.theme.colors.text1};

    &:hover {
      background: inherit;
    }
    cursor: default;
  }

  &[aria-pressed='true'] {
    background: ${({ theme }) => theme.colors.keySubtle};
    color: ${({ theme }) => theme.colors.key};

    &[disabled] {
      background: ${({ theme }) => theme.colors.keySubtle};
      color: ${({ theme }) => theme.colors.keyFocus};
    }
  }
`
