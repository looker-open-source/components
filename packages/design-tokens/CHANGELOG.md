# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/design-tokens





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.26]

### Added

- `AliasColor` to support color aliases previously used via `variant` property
- Added colors: `secondary` & `subdued`
- Added colors: `informAccent`, `positiveAccent`, & `warnAccent`

## [0.9.20]

### Changed

- `theme.transitions` durations are now integers (in milliseconds) rather than strings
- Reduced & consolidated dependencies on `polished` library

## [0.9.17]

### Changed

- Brand font defaults to `Roboto`

## [0.9.15] - 2020-09-21

### Changed

- `theme.fonts.*` now produced with single quotes rather than double quotes

## [0.9.14]

### Changed

- `theme.fonts.*` updates
  - `body` added (now default for most components)
  - `brand` is now used just for `Button*`, `Heading` & `Tabs`
  - Default values changed (now `body='Roboto'`, `brand='Red Hat Display'`, `code='Roboto Mono'`)
- `theme.fontWeights.extraBold` & `theme.fontWeights.light` removed

### Removed

- `prismTheme` is no longer published
- `theme.colors.palette` is no longer available (`palette` _can_ be import from `@looker/design-tokens` but this is a legacy behavior and is not encouraged)

## [0.9.10]

### Added

- Added `xxxxxlarge` to sizes to support updated type ramp.

### Changed

- `FontSizes` and `LineHeights` design tokens updated to match new type ramp spec.
- `neutral` intent color is now defaults to `charcoal500`

## [0.9.7]

### Changed

- `theme` "pressed" colors are more discernable from other stateful colors
- `theme.colors.*Pressed` colors are more discernable from other stateful colors
- `theme.colors.textX` restructured
  - `text1-5` now go from lightest to darkest to match `ui1-5`
  - Reduced number of steps:
    - `text0` is now `text5` (consolidated the former `text0` & `text1`)
    - `text6` is now `text1` (consolidated the former `text6` & `text5`)

## [0.9.6]

### Changed

- `theme.zIndexFloor` now defaults to `1` instead of `undefined` (this helps with compatibility due to Firefox's stacking order eccentricities)

## [0.9.0]

### Added

- `zIndexFloor` added to `theme` (used as base value for Overlay & Modal z-index value:q)

### Removed

- `pseudo` support from design-tokens (prefer using `styled` when pseudo selectors are required)
