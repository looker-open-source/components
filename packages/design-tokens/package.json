{
  "name": "@looker/design-tokens",
  "license": "MIT",
  "version": "0.7.14",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "module": "dist/esm/index.js",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/design-tokens"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build:all": "yarn clean && yarn run build:esm && yarn run build:cjs",
    "build:esm": "ttsc --project tsconfig.build.json --module es2015 --target es6 --outDir dist/esm",
    "build:cjs": "ttsc --project tsconfig.build.json --module commonjs --target es6 --outDir dist/cjs",
    "clean": "rm -Rf ./dist",
    "declaration": "ttsc --project tsconfig.json",
    "develop": "yarn build:es --watch --verbose & yarn declaration --watch"
  },
  "dependencies": {
    "lodash": "^4.17.15",
    "polished": "^3.4.4",
    "styled-system": "^5.1.4"
  },
  "devDependencies": {
    "@types/lodash": "^4.14.149",
    "@types/styled-components": "^4.4.1",
    "@types/styled-system": "^5.1.4",
    "csstype": "^2.6.8",
    "lodash": "^4.17.15",
    "react": "^16.12.0",
    "react-dom": "^16.12.0",
    "styled-components": "^4.4.1",
    "uuid": "^3.4.0"
  },
  "peerDependencies": {
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "styled-components": "^4"
  },
  "gitHead": "05b63c793e74648ca4ada0d1bb9d4cc278256778"
}
