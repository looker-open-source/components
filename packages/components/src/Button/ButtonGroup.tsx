/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import uniqueId from 'lodash/uniqueId'
import xor from 'lodash/xor'
import React, { ChangeEvent, FC, forwardRef, Ref } from 'react'
import styled from 'styled-components'
import { useControlWarn } from '../utils'
import { ButtonItem, ButtonItemLabel } from './ButtonItem'
import { ButtonGroupOrToggleProps, ButtonSet } from './ButtonSet'

const ButtonGroupFactory = forwardRef(
  (
    { onChange, value, ...props }: ButtonGroupOrToggleProps<string[]>,
    ref: Ref<HTMLDivElement>
  ) => {
    const isControlled = useControlWarn({
      controllingProps: ['onChange', 'value'],
      isControlledCheck: () => onChange !== undefined,
      name: 'ButtonGroup',
    })

    function handleChange(e: ChangeEvent<HTMLInputElement>) {
      const newValue = xor(value, [e.target.value])
      onChange && onChange(newValue)
    }

    return (
      <ButtonSet
        {...props}
        ref={ref}
        {...(isControlled
          ? {
              onChange: handleChange,
              value,
            }
          : { name: props.name || uniqueId() })}
      />
    )
  }
)

export const ButtonGroup = styled<FC<ButtonGroupOrToggleProps<string[]>>>(
  ButtonGroupFactory
)`
  ${ButtonItem} {
    height: 36px;
    border-radius: 4px;

    + ${ButtonItem} {
      margin-left: ${(props) => props.theme.space.xxsmall};
    }
  }

  ${ButtonItemLabel} {
    border-style: solid;
    border-width: 1px;
  }
`
