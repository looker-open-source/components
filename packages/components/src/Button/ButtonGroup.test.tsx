/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import '@testing-library/jest-dom/extend-expect'
// import { fireEvent } from '@testing-library/react'
import React from 'react'
import { Simulate } from 'react-dom/test-utils'

import { renderWithTheme } from '@looker/components-test-utils'
import { theme } from '@looker/design-tokens'
import { ButtonGroup } from './ButtonGroup'
import { ButtonItem } from './ButtonItem'

const requiredProps = {
  message: 'Foo',
  onConfirm: jest.fn(),
  title: 'Bar',
}

const optionalProps = {
  cancelLabel: 'Dont Delete',
  confirmLabel: 'Delete',
  message: 'This is permanent',
  onCancel: jest.fn(),
  title: 'Delete the thing?',
}

afterEach(() => {
  requiredProps.onConfirm.mockClear()
  optionalProps.onCancel.mockClear()
})

test('<ButtonGroup/> uncontrolled', () => {
  const onChangeMock = jest.fn()
  const { container, getByLabelText, getByText } = renderWithTheme(
    <form>
      <ButtonGroup name="fruits" label="Fruits">
        <ButtonItem selected>Apples</ButtonItem>
        <ButtonItem onChange={onChangeMock}>Oranges</ButtonItem>
        <ButtonItem>Bananas</ButtonItem>
      </ButtonGroup>
    </form>
  )

  const apples = getByLabelText('Apples')
  const oranges = getByLabelText('Oranges')

  expect(apples).toHaveAttribute('checked')
  expect(oranges).not.toHaveAttribute('checked')
  expect(getByText('Apples')).toHaveStyle(`color: ${theme.colors.key}`)
  expect(getByText('Oranges')).toHaveStyle(`color: ${theme.colors.text3}`)

  expect(getByLabelText('Bananas')).toBeInTheDocument()

  Simulate.change(apples)
  expect(onChangeMock).not.toHaveBeenCalled()

  Simulate.change(oranges)
  expect(onChangeMock).toHaveBeenCalled()

  const form = container.querySelector('form')
  if (form) {
    const formData = new FormData(form)
    expect(formData.get('fruits')).toEqual('Oranges')
  }
})

test('<ButtonGroup/> controlled', () => {
  const onChangeMock = jest.fn()
  const { getByLabelText } = renderWithTheme(
    <ButtonGroup value={['Bananas']} onChange={onChangeMock}>
      <ButtonItem>Apples</ButtonItem>
      <ButtonItem>Oranges</ButtonItem>
      <ButtonItem>Bananas</ButtonItem>
    </ButtonGroup>
  )

  const apples = getByLabelText('Apples')
  const bananas = getByLabelText('Bananas')
  const oranges = getByLabelText('Oranges')

  expect(bananas).toHaveAttribute('checked')
  expect(oranges).not.toHaveAttribute('checked')

  Simulate.change(apples)
  Simulate.change(bananas)
  Simulate.change(oranges)
  expect(onChangeMock).toHaveBeenCalledTimes(3)
  expect(onChangeMock.mock.calls[0][0]).toEqual(['Bananas', 'Apples'])
  expect(onChangeMock.mock.calls[1][0]).toEqual([])
  expect(onChangeMock.mock.calls[2][0]).toEqual(['Bananas', 'Oranges'])
})
