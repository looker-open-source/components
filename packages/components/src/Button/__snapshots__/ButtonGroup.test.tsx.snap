// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup controlled 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  color: #4C535B;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Red Hat Display','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  font-size: 0.875rem;
  height: 36px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  padding: 0 0.75rem;
  -webkit-transition: background 150ms ease;
  transition: background 150ms ease;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:active,
.c3:focus,
.c3:hover {
  background: #E8E5FF;
}

.c3:focus {
  box-shadow: 0 0 0.5px 1px #9785F2;
  outline: none;
}

.c3[disabled] {
  color: #939BA5;
  cursor: default;
}

.c3[disabled]:hover {
  background: inherit;
}

.c3[aria-pressed='true'] {
  background: #F3F2FF;
  color: #6C43E0;
}

.c3[aria-pressed='true'][disabled] {
  background: #F3F2FF;
  color: #9785F2;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 0.875rem;
  text-align: center;
}

.c1 .c2 {
  border: 1px solid #DEE1E5;
  border-radius: 0.25rem;
  margin-right: 0.25rem;
}

.c1 .c2:last-child {
  margin-right: 0;
}

.c1 .c2[aria-pressed='false']:not(:hover) {
  background: #FFFFFF;
}

.c1.wrapping {
  margin: -0.125rem;
}

.c1.wrapping .c2 {
  margin: 0.125rem;
}

<div
  class="wrapping c0 c1"
  role="group"
>
  <button
    aria-pressed="false"
    class="c2 c3"
    type="button"
    value="Apples"
  >
    Apples
  </button>
  <button
    aria-pressed="false"
    class="c2 c3"
    type="button"
    value="Oranges"
  >
    Oranges
  </button>
  <button
    aria-pressed="true"
    class="c2 c3"
    type="button"
    value="Bananas"
  >
    Bananas
  </button>
</div>
`;
